### PR TITLE
Apply Documentation & update requirements with PyPI publish

### PR DESCRIPTION
**Pull request by laurenceyoon**
Updates:

- Apply Sphinx & Readthedocs documentation (https://github.com/pymatchmaker/matchmaker/issues/10)
- add setup guides for pip install pymatchmaker
- removed the usage for madmom for this version. (incompatibility issue)
- set initial package version: 0.1.0
- add github workflow (for publishing to PyPI)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,42 +8,66 @@ Unlike offline alignment methods, for which state-of-the-art implementations are
...
 ## Setup
 
+### Prerequisites
+
+- Available Python version: 3.9, 3.10, 3.11, 3.12 (3.12 recommended)
+- [Fluidsynth](https://www.fluidsynth.org/)
+- [PortAudio](http://www.portaudio.com/)
+
+Please ensure that you've installed the above packages before proceeding.
+You should not install `fluidsynth` using `pip install fluidsynth` as it is not compatible with `matchmaker`.
```
Comment (CarlosCancino-Chacon) [inline README.md:23]: Maybe we can change this sentence to 

```
First, install Fluidsynth, and then install the `pyfluidsynth` Python library. Note that `pyfluidsynth` only provides Python bindings for Fluidsynth; it does not install Fluidsynth itself. Be aware that there is also a `fluidsynth` Python library (without the `py-` prefix), but it is not compatible with `matchmaker`.
```

Comment (CarlosCancino-Chacon) [reply README.md:23]: (this refers to the sentence `You should not install fluidsynth...`)

File: README.md [hunk 2]
```diff
@@ -113,9 +152,9 @@ with AudioStream(processor=feature_processor) as stream:
 
 Matchmaker currently supports the following alignment methods:
 
-- `"dixon"`: On-line time warping algorithm by S. Dixon (2005).
-- `"arzt"`: On-line time warping algorithm adapted from Brazier and Widmer (2020) (based on the work by Arzt et al. (2010))
-- `"hmm"`: Hidden Markov Model-based score follower by Cancino-Chacón et al. (2023), based on the state-space score followers by Duan et al. (2011) and Jiang and Raphael (2020).
+- `"dixon"`: On-line time warping algorithm by S. Dixon (2005). Supports audio input only.
```
Comment (CarlosCancino-Chacon) [inline README.md:155]: I think both OLTW could support MIDI as well, but this requires a little bit more work (which we can address in the next release), maybe we can add "MIDI support will come in the next release"?

File: README.md [hunk 3]
```diff
@@ -113,9 +152,9 @@ with AudioStream(processor=feature_processor) as stream:
...
+- `"arzt"`: On-line time warping algorithm adapted from Brazier and Widmer (2020) (based on the work by Arzt et al. (2010)). Supports audio input only.
+- `"hmm"`: Hidden Markov Model-based score follower by Cancino-Chacón et al. (2023), based on the state-space score followers by Duan et al. (2011) and Jiang and Raphael (2020). Supports MIDI input only.
```
Comment (CarlosCancino-Chacon) [inline README.md:157]: And the same for this one (but say "Audio support will come in the next release")

**Review by CarlosCancino-Chacon — CHANGES_REQUESTED** [review]
Hi @laurenceyoon! Thank you for the pull request. I think it looks good. I only pointed out to a couple of minor changes in the text.

If you agree with them, we can merge the Pull request (I marked this as "Request changes" for the moment)

File: README.md (hunk 1 above)
Comment (laurenceyoon) [reply README.md]: Thanks! Suggestions are updated here: https://github.com/pymatchmaker/matchmaker/pull/13/commits/7ac5682d9676603a376419e633587e7386907c71

File: README.md (hunk 2 above)
Comment (laurenceyoon) [reply README.md]: https://github.com/pymatchmaker/matchmaker/pull/13/commits/7ac5682d9676603a376419e633587e7386907c71 👍

File: README.md (hunk 3 above)
Comment (laurenceyoon) [reply README.md]: https://github.com/pymatchmaker/matchmaker/pull/13/commits/7ac5682d9676603a376419e633587e7386907c71 updated!

**Review by CarlosCancino-Chacon — APPROVED** [review]

**Pull request merged**